lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectDatumWriter.java

```diff
@@ -81,8 +81,7 @@ protected void writeArray(Schema schema, Object datum, Encoder out) throws IOExc
       out.writeArrayStart();
       switch (type) {
       case BOOLEAN:
-        if (elementClass.isPrimitive())
```

Comment:
Removed because there is already a check that the elementClass is a primitive one at line 79.